File: source/jugglinglab/jml/VelocityRef.java
// VelocityRef.java
//
// Copyright 2019 by Jack Boyce (jboyce@gmail.com)

package jugglinglab.jml;

import jugglinglab.util.*;
import jugglinglab.path.*;


public class VelocityRef {
    protected Path  pp;
    protected boolean   start;

    public VelocityRef(Path pp, boolean start) {
        this.pp = pp;
        this.start = start;
    }

    public Coordinate getVelocity() {
        if (start)
            return pp.getStartVelocity();
        else
            return pp.getEndVelocity();
    }
}

File: composeApp/src/commonMain/kotlin/jugglinglab/layout/VelocityRef.kt
//
// VelocityRef.kt
//
// Copyright 2002-2026 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.layout

import jugglinglab.path.Path
import jugglinglab.util.Coordinate

class VelocityRef(val pp: Path, val source: Int) {
    val velocity: Coordinate
        get() = if (this.source == VR_THROW) pp.startVelocity else pp.endVelocity

    companion object {
        var VR_THROW: Int = 0
        var VR_CATCH: Int = 1
        var VR_SOFTCATCH: Int = 2
    }
}
